include/drivers/edac.h

```diff
@@ -109,48 +93,69 @@ static inline int edac_inject_set_param1(const struct device *dev,
  * Get first error injection parameter value.
  *
  * @param dev Pointer to the device structure
- * @return Injection parameter param1
+ * @param value Pointer to the first injection parameter
+ *
+ * @retval -ENOTSUP if not supported
+ * @retval 0 on success, error code otherwise
  */
-static inline uint64_t edac_inject_get_param1(const struct device *dev)
+static inline int edac_inject_get_param1(const struct device *dev,
+					 uint64_t *value)
 {
 	const struct edac_driver_api *api =
 		(const struct edac_driver_api *)dev->api;
 
-	return api->inject_get_param1(dev);
+	if (api->inject_get_param1 == NULL) {
+		return -ENOTSUP;
```

Comment:
Sorry, I should have noticed this detail earlier. Whenever an API isn't implemented the right error is `ENOSYS` and not `ENOTSUP`. See here: https://github.com/zephyrproject-rtos/zephyr/blob/master/doc/reference/api/design_guidelines.rst#return-codes

Reply:
@jhedberg is `ENOSYS` is for mandatory API? This is optional API.

Reply:
@finikorg I don't think it makes any difference. Grep for ENOSYS under include/drivers to see how it's used.